transcom/mymove
# MAIN - B-22083 AOA Packet Won't Download Due to Deleted Order Record

## [B-22083](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-22083)

INT PR: https://github.com/transcom/mymove/pull/14365

## Summary

This fixes the issue where the AOA packet and SSW would not download when a uploaded order had been deleted. The issue was due to pop query not filtering out soft deleted files correctly when fetching docs to build pdfs. Solved by filtering out the userUploads with `deleted_at`. Tried to filter it using pop, but could not get it it work.

### How to test

1. As a customer -> Create a PPM move and choose `yes` when asked `Would you like to request an advance on your incentive?` in Avances page and put in any amount.
2. As a counselor -> Click edit the shipment and continue to 2nd page to approve the AOA. Edit orders and fill in the required entries -> submit the move.
3. `Download AOA Paperwork (PDF)` link should appear in the shipment card. Click and verify that it downloads without an error.
4. As the customer -> upload ppm docs and submit.
5. As the closeout counselor -> review and approve all the docs. Verify you can download the payment packet after without an error.

## Screenshots
![image](https://github.com/user-attachments/assets/f3453823-c03f-4ddd-9e78-05c2cdfa027c)
